### PR TITLE
Release @google-cloud/pubsub v0.25.0

### PR DESCRIPTION
**Pull request by callmehiphop**
This pull request was generated using releasetool.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,42 @@
...
+
+02-20-2019 10:35 PST
+
+### New Features
+- refactor(ts): improve TypeScript types ([#482](https://github.com/googleapis/nodejs-pubsub/pull/482))
+- refactor(typescript): noImplicityAny for snapshot.ts and publisher.ts ([#457](https://github.com/googleapis/nodejs-pubsub/pull/457))
+
+### Bug Fixes
+- fix: ignore messages that come in after close ([#485](https://github.com/googleapis/nodejs-pubsub/pull/485))
+- fix: throw on invalid credentials and update retry config ([#476](https://github.com/googleapis/nodejs-pubsub/pull/476))
```
Comment (crwilcox) [inline CHANGELOG.md]: I am digging into this particular change. it seems odd that so many things changed from non-idempotent to idempotent. If we can hold this for just a bit that would be super.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,42 @@
 
 [1]: https://www.npmjs.com/package/nodejs-pubsub?activeTab=versions
 
+## v0.25.0
```
Comment (JustinBeckwith) [inline CHANGELOG.md]: doesn't this release having breaking changes?

Comment (callmehiphop) [reply CHANGELOG.md]: Maybe this was a mistake, I'm not really sure how to I should be versioning for TypeScript types. I assumed this was a feature, albeit only for TypeScript users, so I went minor.

Comment (callmehiphop) [reply CHANGELOG.md]: Oh wait, I misunderstood, the retries are breaking, you're right

Comment (JustinBeckwith) [reply CHANGELOG.md]: Yes!  Was just saying - we should call that out :)

File: CHANGELOG.md (hunk 1 above)
Comment (callmehiphop) [reply CHANGELOG.md]: @jadekler do you have input here? I want to say I remember you telling us this change was coming (apologies if I'm misremembering).

File: CHANGELOG.md (hunk 2 above)
Comment (jadekler) [reply CHANGELOG.md]: +1 to calling it out, but we're not doing major semver bumps elsewhere if that's part of the question. The "breaking"-ness of the change should be more akin to bug/unintended behavior fixing.

As to calling it out, maybe add something like "If you experience usability concerns, please file an issue at <link>."

Comment (callmehiphop) [reply CHANGELOG.md]: I made a change to include a note about it. I didn't go with the usual **BREAKING** headline, I feel like that should be reserved for surface changes. Let me know if you disagree!

**Review by JustinBeckwith — APPROVED** [review]

**Comment by JustinBeckwith** [comment]
@callmehiphop should I be worried about the failing system tests....?

**Comment by callmehiphop** [comment]
@JustinBeckwith great question. I'm less concerned about the grpc-js failures, but I am curious about the sample failure. Its the same test failing pretty consistently via Kokoro and I hadn't seen this particular failure pop up in a little while. Based on the assertion of that test, my best guess is that `setTimeout` is firing much earlier than it should be, do we know if there's any funkiness with timing functions on Kokoro?

**Comment by JustinBeckwith** [comment]
Yeah - please get the samples tests passing reliably before we ship this out.

**Comment by alexander-fenster** [comment]
Please ignore grpc-js tests failures, it's likely has nothing to do with Pub/Sub code, so when gax is fixed, those tests will start passing. I'm looking into it in gax.

**Comment by jkwlui** [comment]
> Based on the assertion of that test, my best guess is that `setTimeout` is firing much earlier than it should be, do we know if there's any funkiness with timing functions on Kokoro?

@callmehiphop I'm not aware of any issue with `setTimeout` on Kokoro. We could eliminate some possibilities by trying to run the samples test in a docker container. The image used in Kokoro is based on `node:8`.

**Comment by callmehiphop** [comment]
@jkwlui that's what I've been trying to do, I took it a step further and deployed my docker image to GCE, but I haven't managed to get the same failure there.

**Comment by callmehiphop** [comment]
Update on my testing, I have managed to reproduce this several times now, but not consistently. My logs tell me that my hunch was correct and that calling `setTimeout(func, 1000)` can fire as soon as 800ms.

I think we may want to explore the idea of using a timing library as opposed to straight up `setTimeout`.

**Comment by JustinBeckwith** [comment]
Is the setTimeout causing trouble here code in a test, or is it in library code?  If it's test code that's misbehaving, I'm supportive of tweaking that code to get the test passing.  

On the specific issue with setTimeout - if the issue was it not invoking the callback within the allotted time, I'd say yeah.  But I'd it's firing *early* - that's a huge bug in node, and we should engage @ofrobots and the node team.

**Comment by callmehiphop** [comment]
@JustinBeckwith its [library code](https://github.com/googleapis/nodejs-pubsub/blob/master/src/publisher.ts#L209). We use a `setTimeout` to buffer messages and send them as a batch. 

As far as calling it a bug in Node.. I'm not sure. It is pretty well known that `setTimeout` can be jittery, but 200ms seems very high.

**Comment by callmehiphop** [comment]
I was incorrect about `setTimeout` being jittery, thanks @alexander-fenster for figuring out the issue.

**Pull request merged**